gitlab/client.py

```diff
@@ -29,8 +29,9 @@
 from gitlab import utils
 
 REDIRECT_MSG = (
-    "python-gitlab detected an http to https redirection. You "
-    "must update your GitLab URL to use https:// to avoid issues."
+    "python-gitlab detected a {status_code} ({reason!r}) redirection. You must update "
+    "your GitLab URL to the correct URL to avoid issues. The redirection was from: "
+    "{source!r} to {target!r}"
```

Comment:
Any reason for this `!r`. Wouldn't it work without? What does it do exactly here?

Reply:
The `!r` just quote it for us automatically. So the message looks like this:

```
python-gitlab detected a 302 ('Moved Temporarily') redirection. You must update your 
GitLab URL to the correct URL to avoid issues. The redirection was 
from: 'https://gitlab.labs.nic.cz/api/v4/user/status'
to 'https://gitlab.nic.cz/api/v4/user/status'
```

Note: I manually word-wrapped it above so it would be easier to read.

Here is raw output from a test run:
```
gitlab.exceptions.RedirectError: python-gitlab detected a 302 ('Moved Temporarily') redirection. You must update your GitLab URL to the correct URL to avoid issues. The redirection was from: 'https://gitlab.labs.nic.cz/api/v4/user/status' to 'https://gitlab.nic.cz/api/v4/user/status'
```